### ISSUE TITLE
Feature request: Support DnD when meeting without sharing

### ISSUE THREAD

**Issue by paulbatessonos** [bdkjones/MuzzleApp#44]
We were just discussion distractions when running meeting, it would be awesome to have Muzzle supported DnD for joining any meeting not just screen sharing. 

**Potential Interface**
* Menu bar toggle for **Muzzle When >** _Screen Sharing_   | _Participating in a Meeting_ 
* Preferences drop down: **Muzzle When >** _Screen Sharing_   | _Participating in a Meeting_

**Comment by SteveIsles** [comment]
Yes, this would be extremely useful.  Perhaps merely detecting that a particular app is running (Web-Ex) would be sufficient to activate DND.

**Comment by LuciMcK** [comment]
I'd love this option, too.

**Comment by siapplebaum** [comment]
I agree.

**Comment by bdkjones** [comment]
Well, the problem with this approach is that most of these apps can be running without screensharing. Skype, for instance, sits idle in the background. Or many meeting services that are really just chrome plugins. Can’t turn on DND whenever Skype and Chrome are running. There may be a handful of apps where this is useful, but in general it won’t catch too many cases

**Comment by siapplebaum** [comment]
Thanks for the quick response! I'm wondering if we could trigger on using the microphone.

**Comment by bdkjones** [comment]
Unlikely, because on Mojave the microphone is always in-use for “hey siri” on Macs that support it.

**Comment by bdkjones** [comment]
I’m also fairly sure that detecting whether an app is using the microphone would require a kernel extension, because that’s all very locked down for obvious security reasons. Unix won’t allow one process to figure out which other process is using a certain resource, unless you wander down into the kernel and figure it out there.

**Comment by siapplebaum** [comment]
I agree. I wish I had a better idea.
